torch/fx/passes/_tensorify_python_scalars.py

```diff
@@ -207,6 +211,24 @@ def _sympy_interp(expr: sympy.Expr) -> MetaProxy:
                         node, tracer=tracer, fake_mode=fake_mode
                     )
 
+            # Specialize all dimensions that contain symfloats. Here's
+            # an example test that requires this:
+            # PYTORCH_OPINFO_SAMPLE_INPUT_INDEX=4 tlp python test/inductor/test_torchinductor_opinfo.py TestInductorOpInfoCUDA.test_comprehensive_nn_functional_interpolate_bicubic_cuda_float32 # noqa: B950
+            val = node.meta.get("val")
+            if isinstance(val, FakeTensor):
+                for dim in val.shape:
+                    if isinstance(
+                        dim,
+                        (torch.SymFloat, torch.SymInt, torch.SymBool),
+                    ):
+                        for s in dim.node.expr.free_symbols:
+                            name = str(s)
+                            if symbol_is_type(
+                                s, SymT.FLOAT
+                            ) and not TensorifyState.should_specialize(name):
+                                TensorifyState.specialize(name)
+                                should_restart = True
```

Comment:
This is good, although it's probably worth elaborating in a little more detail why we choose to spec here. In principle, we could support float input that is used to do size compute. The problem is that we don't actually want to tensorify the compute in this case, which means need support for codegen all the float! (Maybe if someone comes to us with a legitimate use case, this restriction could be lifted, perhaps.)